### DISCUSSION TITLE
Point tech avec équipe SRE

### DISCUSSION THREAD

**Discussion by carolineBda** [SocialGouv/code-du-travail-numerique#3856] [Technique]
### [Organisation SRE / Support / DevOps](https://socialgouv.github.io/support)
- présentation de leur équipe
- par exemple recherche-entreprises
- canal de communication avec eux (dans les 2 sens)

### Pipeline
- [migration Github](https://github.com/SocialGouv/code-du-travail-numerique/issues/3760) => pair programing avec eux / session de migration d'une tache
- verif tous les projets

### Alertes 
- mattermost
- site health check
- support du site (off hours)

**Comment by m-maillot** [comment]
Concernant l'organisation, le document est vraiment complet à mon avis. Ce qu'il manque c'est l'équipe. Je sais qu'il y a Gary et Julien mais pendant un moment je ne savais pas pour Gary.
Ce qui peut être intéressant quand même c'est de comprendre l'implication de l'équipe dans le projet. Sur la doc, ils sont une équipe transverse qui intervient lors de besoin. Par contre j'ai l'impression qu'ils utilisent des applications en POC et c'est pour ça qu'il peut y avoir des PR sur un projet sans issue associée de notre part.

Pour la pipeline, je pense qu'une session avec eux et d'établir une "roadmap" de la migration avec également les points plus complexes comme les cron jobs gourmands, les jobs de release...
Concernant la communication, du coup comme l'issue existe c'est plus à nous de faire attention aux nouvelles issues (routine du matin ?) pour en discuter.

Pour l'alerting, le site nous donne pas mal d'information. Par contre, c'est vrai que ça pourrait être bien de lister les sources d'alerting et comment on doit les traiter (même si ça reste interne à note projet).

**Reply by carolineBda** [reply]
Pour la communication, même s'il y avait l'issue, on peut juste valider que c'est comme ça que ça se passe

**Reply by carolineBda** [reply]
Effectivement la doc est top ! J'ai appris plein de trucs

**Comment by carolineBda** [comment]
Si on a du temps j'ai bien envie de demander pourquoi le choix de Gherkin (juste par curiosité)

**Reply by m-maillot** [reply]
On peut en profiter pour demander sur les TU. Car la section est vide et on est en plein dedans. A savoir que l'on utilise JEST et on voudrait tester nos composants react mais pas des snapshots.

**Reply by maxgfr** [reply]
Oui très juste, chaud de voir comment ça se passe le testing sur les autres startup !

**Reply by revolunet** [reply]
Pour gherkin, un des arguments pour ce choix c'était la possibilité pour les PO de définir eux même les tests; dans les faits ce sont les devs qui ont écrit les tests

**Comment by m-maillot** [comment]
Suite à la discussion, il faut planifier 3 points avec la team SRE:

 * Initialisation du workflow Github sur frontend (mob programming)  + Roadmap de la migration (points chauds -> test API, branches de tests...)
 * Présentation du produit Grafana + les outils pour aider à debug (notamment les alertes dans mattermost)
 * Passe sur les libs satellites au projet (https://github.com/SocialGouv/cdtn-admin/blob/master/README-dev.md#linked-repositories). Présentation rapide + archi + point technique particulier + vérification workflow est bien sur Github.
 
 Ces points sont à planifier par l'équipe CDTN selon les dispos.

@carolineBda @maxgfr @gary-van-woerkens @revolunet
